test_symb_tools.py

```diff
@@ -64,17 +64,17 @@ def test_perform_time_deriv3(self):
         test to provide deriv_symbols
         """
 
-        a, b, ad, bd, add, bdd = sp.symbols("a, b, ad, bd, add, bdd")
+        a, b, adot, bdot, addot, bddot = sp.symbols("a, b, adot, bdot, addot, bddot")
 
         f1 = 8*a*b**2
-        f1d = st.perform_time_derivative(f1, (a, b), (ad, bd))
-        self.assertEquals(f1d, 8*ad*b**2 + 16*a*bd*b)
+        f1d = st.perform_time_derivative(f1, (a, b), (adot, bdot))
+        self.assertEquals(f1d, 8*adot*b**2 + 16*a*bdot*b)
 
-        f1d_2 = st.perform_time_derivative(f1, (a, b), (ad, bd)+ (add, bdd),
+        f1d_2 = st.perform_time_derivative(f1, (a, b), (adot, bdot)+ (addot, bddot),
                                            order=2)
 
-        f1d_2_altntv = st.perform_time_derivative(f1d, (a, b, ad, bd),
-                                                 (ad, bd)+ (add, bdd) )
+        f1d_2_altntv = st.perform_time_derivative(f1d, (a, b, adot, bdot),
+                                                 (adot, bdot)+ (addot, bddot) )
         self.assertEquals(f1d_2, f1d_2_altntv)
 
```

Comment:
Bei Symbolen die hinten keine Nummer haben stimmt glaube ich noch irgedwas nicht:

```python
    @unittest.expectedFailure
    def test_perform_time_deriv4(self):
        # test higher order derivatives

        a, b = sp.symbols("a, b")

        f1 = 8*a*b**2

        res_a1 = st.perform_time_derivative(f1, (a, b), order=5)

        a_str = 'a adot addot adddot a_d4 a_d5'
        b_str = a_str.replace('a', 'b')

        expected_symbol_names = a_str.split() + b_str.split()

        res_list =  [e in res_a1 for e in expected_symbol_names]

        #IPS() # this will fail
        self.assertTrue( all(res_list) )

        l1 = len( res_a1.atoms(sp.Symbol) )
        self.assertEqual(len(expected_symbol_names), l1)
```